src/FFT.c

```diff
@@ -0,0 +1,112 @@
+#include "FFT.h"
+#include <math.h>
+#include <stdlib.h>
+
+/**
+ * This method implements bit reversal needed by the FFT.
+ * Uses the Gold and Rader's bit reversal algorithm
+ * @param length The length of the input vectors. Must be a power of 2
+ * @param realArray 1D array containing the real part of the incoming vector.
+ * @param imaginaryArray 1D array containing the imaginary part of the incoming vector.
+ */
+static void bitReversal(unsigned length, fft_real* realArray, fft_real* imaginaryArray) {
```

Comment:
I would even call the function goldRaderBitTraversal since the name is ambiguous